src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -672,71 +678,82 @@ const static auto formats = {
     std::locale(std::locale::classic(),
                 new boost::local_time::local_time_input_facet("%Y-%m-%d %H:%M:%s%ZP")),
 };
-const static boost::posix_time::ptime epoch(boost::gregorian::date(1970, 1, 1));
-const static boost::posix_time::ptime max_date(boost::gregorian::date(2030, 1, 1));
 
-static std::chrono::milliseconds parseDateAsMillis(
-    const std::string& datetime, const boost::posix_time::ptime& default_time = epoch) {
-    boost::posix_time::ptime utc_time{boost::posix_time::not_a_date_time};
-    auto millis = 0LL;
+class DateToIntGenerator : public Generator<int64_t> {
+public:
+    explicit DateToIntGenerator(UniqueGenerator<bsoncxx::types::b_date>& generator) {
+        std::swap(_generator, generator);
+    }
+    int64_t evaluate() override {
+        auto date = _generator->evaluate();
+        auto millis = date.to_int64();
+        // BOOST_LOG_TRIVIAL(warning) << " DateToIntGenerator " << millis;
+        return millis;
+    }
 
-    // Empty datetime is 0.
-    if (!datetime.empty()) {
-        for (const auto& format : formats) {
-            std::istringstream date_stream{datetime};
-            date_stream.imbue(format);
-            boost::local_time::local_date_time local_date{boost::local_time::not_a_date_time};
-            if (date_stream >> local_date) {
-                utc_time = local_date.utc_time();
+protected:
+    UniqueGenerator<bsoncxx::types::b_date> _generator;
+};
 
-                millis = (utc_time - epoch).total_milliseconds();
-                break;
-            }
-        }
+class DoubleToIntGenerator : public Generator<int64_t> {
+public:
+    explicit DoubleToIntGenerator(UniqueGenerator<double>& generator) {
+        std::swap(_generator, generator);
+    }
+    int64_t evaluate() override {
+        auto millis = (long long)_generator->evaluate();
```

Comment:
so `_gen->eval()` returns `double` which we cast to `long long` but the expected return type is `int64`? I think this is working by casting twice which is why you needed the separate `auto millis` declaration.

I think you can use a single cast:

```c++
int64_t evaluate() override {
  return (int64_t) _gen->eval();
}
```

Also consider naming this `CastDoubleToInt64Generator` or similar.